doc/advanced/content/pcl_style_guide.rst

```diff
@@ -317,64 +317,35 @@ indentation level of the class body and members affected by these qualifiers are
 2.6. Automatic code formatting
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The following set of rules can be automatically used by various different IDEs,
-editors, etc.
+We currently use clang-format as the tool for auto-formatting our C++ code. The style files which were previously distributed should now be considered deprecated.
 
-2.6.1. Emacs
-""""""""""""
+The style rules mentioned in this document are enforced via `PCL's .clang-format file
+<https://github.com/PointCloudLibrary/pcl/blob/master/.clang-format>`_.
 
-You can use the following `PCL C/C++ style file
-<https://raw.githubusercontent.com/PointCloudLibrary/pcl/master/doc/advanced/content/files/pcl-c-style.el>`_,
-download it to some known location and then:
+For the integration of clang-format with various text editors and IDE's, refer `this
+<https://clang.llvm.org/docs/ClangFormat.html>`_.
 
-* open .emacs 
-* add the following before any C/C++ custom hooks
+Details about the style options used can be found `here
+<https://clang.llvm.org/docs/ClangFormatStyleOptions.html>`_.
 
-::
+2.6.1. Script usage
+"""""""""""""""""""
 
-   (load-file "/location/to/pcl-c-style.el")
-   (add-hook 'c-mode-common-hook 'pcl-set-c-style)
+PCL also has a supporting `format script
+<https://github.com/PointCloudLibrary/pcl/blob/master/.dev/format.sh>`_, 
+which will run the clang-format on the whitelisted directories.
 
-2.6.2. Uncrustify
-"""""""""""""""""
+Sample command line usage from the PCL source directory:
 
-You can find a semi-finished config for `Uncrustify <http://uncrustify.sourceforge.net/>`_ `here
-<http://dev.pointclouds.org/attachments/download/537/uncrustify.cfg>`_
+.. code-block:: shell
 
-2.6.3 Eclipse
-"""""""""""""
+   $ sh ./.dev/format.sh `which clang-format` ./
```

Comment:
I would rather not expose this. Users should use the make target.

Reply:
Done.